mmdet/models/detectors/kd.py

```diff
@@ -0,0 +1,76 @@
+import torch
+
+from ..builder import DETECTORS
+from .single_stage import SingleStageDetector
+
+
+@DETECTORS.register_module()
+class KnowledgeDistillationSingleStageDetector(SingleStageDetector):
+
+    def __init__(self,
+                 backbone,
+                 neck,
+                 bbox_head,
+                 teacher_config='',
+                 teacher_model='',
+                 train_cfg=None,
+                 test_cfg=None,
+                 pretrained=None):
+        super().__init__(backbone, neck, bbox_head, train_cfg, test_cfg,
+                         pretrained)
+
+        from mmdet.apis.inference import init_detector
+
+        self.teacher_model = init_detector(
+            teacher_config, teacher_model, device=torch.cuda.current_device())
```

Comment:
`torch.cuda.current_device()` is a hardcode, since users may use cpu devices too. 
Now the `detector.cuda()` does not put the teacher model into cuda since `self.teacher_model` is not a submodule any more. Therefore, we may need to relload the `cuda()` method of nn.Module. 

```python
    def cuda(self, device=None):
        """It also puts the teacher model to cuda"""
        self.teacher_model.cuda(device=device)
        return super().cuda(device=device)
```

Reply:
Besides, the BN/dropout train mode of the teacher also need to be considered, even though the original paper does not need to since the BN layers are all fixed. 
Suggestion: Reload the `train() method of nn.Module:
```python
    def train(self, mode=True):
        """Set the same train mode for teacher and student model"""
        if self.eval_teacher:
            self.teacher_model.train(False)
        else:
            self.teacher_model.train(mode)
        super().train(mode)
```